### ISSUE TITLE
XML validation doesn't clean up after itself

### ISSUE THREAD

**Issue by shorowit** [NREL/OpenStudio#4761]
Issue overview
--------------
I am using OpenStudio 3.5's XSD/Schematron validation in BEopt, where we may end up validating thousands of HPXML files during an optimization of EnergyPlus simulations.

The computer ended up crashing because it ran out of disk space. It looks like every individual OpenStudio validation produces a 1MB temp directory:

![image](https://user-images.githubusercontent.com/5861765/206790321-b954b5c8-e069-4656-98f5-4abc7c574dff.png)

After running just a few BEopt optimizations, there were over 10k directories taking up 12 GB of space.

Can OpenStudio clean up the temp dir after validation?

@jmarrec

**Comment by shorowit** [comment]
Even better, could it avoid creating these files and do everything in memory?

**Comment by jmarrec** [comment]
We can try to clean up. But no, we cannot do it in memory unfortunately (my memory is strong on the fact we can't, more blurry on which lib is at fault but i can pull it up if you need proof)

**Comment by jmarrec** [comment]
Perhaps we should rather look into providing facilities to reuse the result of the XLST transform instead, I can't recall if thats doable or not but can investigate

**Comment by shorowit** [comment]
When I first took advantage of this capability, I looked at reusing the xslt but found that it was only 0.05 seconds faster (see [here](https://github.com/NREL/OpenStudio-HPXML/pull/1127)). So I determined it wasn't worth the extra complexity on my end. I'd be fine if you just clean up the temp dir, call it a day, and move on to other priorities.

I do have [a workaround](https://github.com/NREL/OpenStudio-HPXML/pull/1246) in place on our end, since we can't afford to wait for the next OS release.
